vllm/worker/model_runner.py

```diff
@@ -788,6 +788,53 @@ def list_loras(self) -> Set[int]:
             raise RuntimeError("LoRA is not enabled.")
         return self.lora_manager.list_loras()
 
+    @torch.inference_mode()
+    def warmup_prefix_attn(self, kv_caches: List[torch.Tensor]) -> None:
+        """Prefix attention uses a triton jit.
+
+        In our profile_run() step, we profile with random data, so the case
+        with a cache hit is not executed. The triton JIT is generated on
+        the fly, so the first call to context_attention_fwd will take
+        ~3s to process. Without this warmup, this JIT will occur on the hot
+        path.
+
+        In this case, we make 2 sequences. Sequence 0 runs prompt_fwd with
+        self.block_size + 1 tokens, filling up physical block 1. Sequence
+        1 then runs the same prompt, but with metadata that block 1 is 
+        computed. This thus triggers context_attention_fwd and generates
+        the code.
+        """
+        NUM_ITERATIONS = 10
```

Comment:
I feel one iteration should be good enough?

Reply:
I thought so too, but empirically seemed that I needed more than 1 run to make timing stable, so I just picked 10. 
This takes <1s so its not impactful to UX, but agree the code is a bit silly

Let me do some more experiments.